interactions/api/models/channel.py

```diff
@@ -95,6 +102,181 @@ class ThreadMember(ClientSerializerMixin):
+        check: Optional[Callable[["Message"], bool]] = None,
+        reverse: Optional[bool] = False,
+    ):
+        super().__init__(obj, _client, maximum=maximum, start_at=start_at, check=check)
+
+        from .message import Message
+
+        if reverse and start_at is MISSING:
+            raise LibraryException(
+                code=12,
+                message="A message to start from is required to go through the channel in reverse.",
+            )
+
+        if reverse:
+            self.before = MISSING
+            self.after = self.start_at
+        else:
+            self.before = self.start_at
+            self.after = MISSING
+
+        self.objects: Optional[List[Message]]
+
+    async def get_first_objects(self) -> None:
+        from .message import Message
+
+        limit = min(self.maximum, 100)
+
+        if self.maximum == limit:
+            self.__stop = True
+
+        if self.after is not MISSING:
+            msgs = await self._client.get_channel_messages(
+                channel_id=self.object_id, after=self.after, limit=limit
+            )
+            msgs.reverse()
+            self.after = int(msgs[-1]["id"])
+        else:
+            msgs = await self._client.get_channel_messages(
+                channel_id=self.object_id, before=self.before, limit=limit
+            )
+            self.before = int(msgs[-1]["id"])
+
+        if len(msgs) < 100:
+            # already all messages resolved with one operation
+            self.__stop = True
+
+        self.object_count += limit
+
+        self.objects = [Message(**msg, _client=self._client) for msg in msgs]
+
+    async def flatten(self) -> List["Message"]:
+        """returns all remaining items as list"""
+        return [item async for item in self]
+
+    async def get_objects(self) -> None:
+        from .message import Message
+
+        limit = min(50, self.maximum - self.object_count)
+
+        if self.after is not MISSING:
+            msgs = await self._client.get_channel_messages(
+                channel_id=self.object_id, after=self.after, limit=limit
+            )
+            msgs.reverse()
+            self.after = int(msgs[-1]["id"])
+        else:
+            msgs = await self._client.get_channel_messages(
+                channel_id=self.object_id, before=self.before, limit=limit
+            )
+            self.before = int(msgs[-1]["id"])
+
+        if len(msgs) < limit or limit == self.maximum - self.object_count:
+            # end of messages reached again
+            self.__stop = True
+
+        self.object_count += limit
+
+        self.objects.extend([Message(**msg, _client=self._client) for msg in msgs])
+
+    async def __anext__(self) -> "Message":
+        if self.objects is None:
+            await self.get_first_objects()
+
+        try:
+            obj = self.objects.pop(0)
+
+            if self.check:
+
+                _res = self.check(obj)
```

Comment:
Can you add support for async checks? It should just be `if isawaitable(res): res = await res`